akka-remote/src/main/scala/akka/remote/AckedDelivery.scala

```diff
@@ -99,6 +99,8 @@ case class AckedSendBuffer[T <: HasSequenceNumber](
    * @return An updated buffer containing the remaining unacknowledged messages
    */
   def acknowledge(ack: Ack): AckedSendBuffer[T] = {
+    if (ack.cumulativeAck > maxSeq)
+      throw new IllegalStateException(s"Highest SEQ so far was $maxSeq but cumulative ACK is ${ack.cumulativeAck}")
```

Comment:
`IllegalArgumentException` is thrown from `buffer`, shouldn't we use same type of exception?